src/library/Connect/WorkloadInformationProvider.cs

```diff
@@ -117,6 +121,27 @@ internal class WorkloadInformationProvider : IWorkloadInformationProvider
             }
         }
 
+        //
+        public List<int> GetPortsToIgnoreFromAnnotations(V1ServiceList serviceNameSpace) {
+            var portsToIgnore = new List<int>();
+            foreach(var item in serviceNameSpace.Items) {
+                if (item.Metadata != null && item.Metadata.Annotations != null) {
+                    if (item.Metadata.Annotations.TryGetValue("bridgetokubernetes/ignore-ports", out var ports)) {
+                        var allPorts = ports.Split(",");
+                        foreach(var port in allPorts) {
+                            try {
+                                portsToIgnore.Add(int.Parse(port));
+                            } catch {
+                                throw new UserVisibleException(this._operationContext, $"bridgetokubernetes/ignore-ports configuration value {port} is invalid. It should be a comma separated list of integer ports");
+                            }
+                        }
+                    }
+                }
+            }
+            _log.Info("Per bridgetokubernetes/ignore-ports user config the following ports will be ignored: {0}", portsToIgnore.Count > 0 ? portsToIgnore.ToString() : "none");

```

Comment:
hi Vidya, just a suggestion to improvise this, you can use where (which filters items based on some conditions) and avoid extra for each (nested) by using string explicitly and toList(). Thanks. I didn't test this, please make sure to test this if you are planning to use them :).
```suggestion
            _public List<int> GetPortsToIgnoreFromAnnotations(V1ServiceList serviceNameSpace) {
            var portsToIgnore = new List<int>();
            // only filter the services which has this annotation present
            List<V1Service> filteredList = serviceNameSpace.Items.Where(item => item.Metadata != null && item.Metadata.Annotations != null 
                && item.Metadata.Annotations.ContainsKey("bridgetokubernetes/ignore-ports")).ToList();

            // loop through them
            filteredList.ForEach(item =>
            {
                if (item.Metadata.Annotations.TryGetValue("bridgetokubernetes/ignore-ports", out string ports))
                {
                    if (ports.Length > 0)
                    {
                        try {
                            // this avoids extra for each loop
                            portsToIgnore = ports.Split(",").ToList();
                        }
                        catch
                        {
                            throw new UserVisibleException(this._operationContext, $"bridgetokubernetes/ignore-ports configuration value {port} is invalid. It should be a comma separated list of integer ports");
                        }
                    }
                    
                }

            });
            _log.Info("Per bridgetokubernetes/ignore-ports user config the following ports will be ignored: {0}", portsToIgnore.Count > 0 ? portsToIgnore.ToString() : "none");
            return portsToIgnore;
        }

```